provider/puppet.rb

```diff
@@ -204,10 +206,13 @@ def compile_end2end_tests(output_folder)
     end
 
     def property_body(property)
-      prop_generator = property_map.select { |type, _| property.class <= type }
-      raise "Unknown property type: #{property}" if prop_generator.empty?
-      body = prop_generator.values[0].call(property)
-      "#{body}\n" unless body.nil?
+      lines(indent([
```

Comment:
Aw, geez, I hate to see `lines(indent([(lines(['`.  Is it coming up on the time we fix `lines` Once And For All?

Reply:
That particular example was bad. We don't need that. I changed it to avoid the double lines.